esmvalcore/_recipe/recipe.py

```diff
@@ -227,6 +228,34 @@ def _get_default_settings(dataset):
     return settings
 
 
+def _add_dataset_specific_settings(dataset: Dataset, settings: dict) -> None:
+    """Add dataset-specific settings."""
+    project = dataset.facets["project"]
+    dataset_name = dataset.facets["dataset"]
+    file_suffixes = [Path(file.name).suffix for file in dataset.files]
+
+    # Automatic regridding for native ERA5 data in GRIB format if regridding
+    # step is not already present (can be disabled with facet
+    # automatic_regrid=False)
+    if all(
+        [
+            project == "native6",
+            dataset_name == "ERA5",
+            any(grib_format in file_suffixes for grib_format in GRIB_FORMATS),
+            "regrid" not in settings,
+            dataset.facets.get("automatic_regrid", True),
+        ]
+    ):
+        # Settings recommended by ECMWF
+        # (https://confluence.ecmwf.int/display/CKB/ERA5%3A+What+is+the+spatial+reference#heading-Interpolation)
+        settings["regrid"] = {"target_grid": "0.25x0.25", "scheme": "linear"}
+        logger.debug(
+            "Automatically regrid native6 ERA5 data in GRIB format with the "
+            "settings %s",
+            settings["regrid"],
+        )
+
+
```

Comment:
If we want to add other GRIB data sources later on, wouldn't it be better to not hardcode ERA5 here, but instead have it as an overwriteable default? Or do you only want to regrid ERA5 on default because of the ECMWF recommendations?

Reply:
ERA5 is hardcoded here on purpose. We really only want the automatic regridding in this specific case. If we want to extend it later, we can simply add another `elif` 👍